### ISSUE TITLE
Calling 'kthxbye' while an event is being posted will cause events to be missed

### ISSUE THREAD

**Issue by avnerenv0** [lazywithclass/winston-cloudwatch#117]
Because `lib/cloudwatch-integration.js` line 20, the callback is called, if this comes from `kthkbye`, the transport is silently "closed" while events might still be waiting.

**Comment by lazywithclass** [comment]
`kthxbye` has been added for scenarios like AWS Lambda, when at a certain point in time you are done and want to just exit because all work has been done.

What is your use case?

**Comment by avnerenv0** [comment]
My use case is a short running Docker container, pretty similar to Lambda. I would also like the process to exit at some point, but I'd like to be sure all the log events have been sent.

**Comment by lazywithclass** [comment]
Yes you are right, we should not just randomly discard events, I will think of a possibile solution for this.

**Comment by avnerenv0** [comment]
I opened https://github.com/lazywithclass/winston-cloudwatch/pull/118 for this
